### ISSUE TITLE
\Cmp\Http\RequestFactory::buildBody does not properly encode generated json

### ISSUE THREAD

**Issue by skyellin** [CMProductions/http-client#17]
See vendor/pluggit/http-client/src/RequestFactory.php:146

Note that json_encode is being called before replacing the placeholders with values.  This results in incorrectly encoded json.

To correct, replace values before calling json encode, e.g.:

```php
        if ($this->isJson($headers)) {
            $post = array_combine($post, $values);
            $json = json_encode($post);

            return $json;
        }
```

Note that the above code will fail if the number of values in `$post` and `$values` is different, where as the previous version would only ignore extra placeholders or values, so additional code should be added to avoid those types of failures.

**Comment by danielmartincmp** [comment]
https://github.com/CMProductions/http-client/pull/18 tries to fix the problem.

Also, not sure if we should implement this:
Not include the param when it's placeholder hasn't been replaced (not send in the call from code).
It's either that or sending it empty. Right now the placeholder string is sent.
Probably not related with this issue...

**Comment by hmoragregaCMP** [comment]
Closing issue, it has been fixed on #18 . Thanks guys!
